kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h

```diff
@@ -23,97 +23,105 @@
-            {
-                // Compute the position, the base and the acceleration vector
-                std::vector<array_1d<double, 3>> derivatives;
-                rNurbsCurve.GlobalSpaceDerivatives(
-                    derivatives,
-                    rParameterLocalCoordinates,
-                    2);
-                rResultLocal = derivatives[0];
-
-                // Compute the distance vector between the point and its 
-                // projection on the curve
-                array_1d<double, 3> distance_vector = rResultLocal - rPointGlobal;
-                if (norm_2(distance_vector) < Accuracy)
-                    return true;
-
-                // Compute the residual
-                residual = inner_prod(distance_vector, derivatives[1]);
-                if (std::abs(residual) < Accuracy)
-                    return true;
-
-                // Compute the increment
-                delta_t = residual / (inner_prod(derivatives[2], distance_vector) + pow(norm_2(derivatives[1]), 2));
-
-                // Increment the parametric coordinate
-                rParameterLocalCoordinates[0] -= delta_t;
-
-                // Check if the increment is too small and if yes return true
-                if (norm_2(delta_t*derivatives[1]) < Accuracy)
-                    return true;
-
-                // Check if the parameter gets out of its interval of definition and if so clamp it 
-                // back to the boundaries
-                rNurbsCurve.DomainInterval().IsInside(rParameterLocalCoordinates[0]);
+            // Compute the position, the base and the acceleration vector
+            rGeometry.GlobalSpaceDerivatives(
+                derivatives,
+                rParameterLocalCoordinates,
+                2);
+            rResultLocal = derivatives[0];
+
+            // Compute the distance vector between the point and its 
+            // projection on the curve
+            distance_vector = rResultLocal - rPointGlobal;
+            if (norm_2(distance_vector) < Accuracy)
+                return true;
+
+            // Compute the residual
+            residual = inner_prod(distance_vector, derivatives[1]);
+            if (std::abs(residual) < Accuracy)
+                return true;
+
+            // Compute the increment
+            delta_t = residual / (inner_prod(derivatives[2], distance_vector) + pow(norm_2(derivatives[1]), 2));
+
+            // Increment the parametric coordinate
+            rParameterLocalCoordinates[0] -= delta_t;
+
+            // Check if the increment is too small and if yes return true
+            if (norm_2(delta_t * derivatives[1]) < Accuracy)
+                return true;
+
+            // Check if the parameter gets out of its interval of definition and if so clamp it 
+            // back to the boundaries
+            int check = 1;//rGeometry.ClosestPointLocalSpace(rParameterLocalCoordinates);
+            if (check == 0) {
```

Comment:
Don't understand. What is this doing?

Reply:
well this is yet depending on #8491 . However, I do not want to create too many dependencies. Basically it resets the point to the boundary after leaving it. After the second time it finishes the algorithm as then it is considered that there won't be any convergence.